packages/ai/src/registry/provider-registry.ts

```diff
@@ -171,14 +199,21 @@ class DefaultProviderRegistry<
     id: `${KEY & string}${SEPARATOR}${string}`,
   ): LanguageModelV2 {
     const [providerId, modelId] = this.splitId(id, 'languageModel');
-    const model = this.getProvider(providerId, 'languageModel').languageModel?.(
+    let model = this.getProvider(providerId, 'languageModel').languageModel?.(
       modelId,
     );
 
     if (model == null) {
       throw new NoSuchModelError({ modelId: id, modelType: 'languageModel' });
     }
 
+    if (this.languageModelMiddleware != null) {
+      model = wrapLanguageModel({
+        model,
+        middleware: this.languageModelMiddleware,
+      });
+    }
+
     return model;
```

Comment:
would probably change this into const + ternary return but good to land unless you want to change this